src/gui/utils.cpp

```diff
@@ -146,7 +146,30 @@ void Utils::Gui::openPath(const Path &path)
     const QUrl url = path.data().startsWith(u"//")
         ? QUrl(u"file:" + path.data())
         : QUrl::fromLocalFile(path.data());
+
+#ifdef Q_OS_WIN
+    auto *thread = QThread::create([path]()
+    {
+        if (SUCCEEDED(::CoInitializeEx(NULL, COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE)))
+        {
+            const std::wstring pathWStr = path.toString().toStdWString();
+            PIDLIST_ABSOLUTE pidl = ::ILCreateFromPathW(pathWStr.c_str());
+            ITEMIDLIST idNull = {0};
+            LPCITEMIDLIST pidlNull[1] = {&idNull};
+            if (pidl)
+            {
+                ::SHOpenFolderAndSelectItems(pidl, 0, pidlNull, 0);
```

Comment:
why not this?
```suggestion
                ::SHOpenFolderAndSelectItems(pidl, 0, nullptr, 0);
```

And assuming this change is made, the code now duplicates with `Utils::Gui::openFolderSelect()` and it should be extracted for re-use.

Reply:
> the code now duplicates with `Utils::Gui::openFolderSelect()`

And it's confusing. Shouldn't this function do something different from `Utils::Gui::openFolderSelect()`?

Reply:
It needs to be pidlNull as with nullptr i had unspecified exceptions, also my bad, the cidl value should be 1, so i fixed that too.

I did some tests and checked online sources, its seems this is the way it needs to be to work if it uses that function, but does not select anything, just opens the folder.

The difference is that `Utils::Gui::openFolderSelect()` is used when there is only 1 file in the torrent or the option "Open containing folder" was used from the Content bottom tab.

Other than that `Utils::Gui::openPath` is used to just open the folder and not select anything or as a fallback in `Utils::Gui::openFolderSelect()` in specific cases.

This is how it is organized in the source so i didn't touch what, the goal was to fix the Windows delay, it just works, but the other systems still needs to use `QDesktopServices::openUrl`.